src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/DSA.cs

```diff
@@ -1125,5 +1125,168 @@ public int GetMaxSignatureSize(DSASignatureFormat signatureFormat)
+        }
+
+        /// <summary>
+        /// Imports an encrypted RFC 7468 PEM-encoded private key, replacing the keys for this object.
+        /// </summary>
+        /// <param name="input">The PEM text of the encrypted key to import.</param>
+        /// <param name="passwordBytes">
+        /// The bytes to use as a password when decrypting the key material.
+        /// </param>
+        /// <exception cref="ArgumentException">
+        /// <paramref name="input"/> does not contain a PEM-encoded key with a recognized label.
+        ///   <para>
+        ///       -or-
+        ///   </para>
+        /// <paramref name="input"/> contains multiple PEM-encoded keys with a recognized label.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///  The password is incorrect.
+        ///   <para>
+        ///       -or-
+        ///   </para>
+        ///  The base-64 decoded contents of the PEM text from <paramref name="input" />
+        ///  do not represent an ASN.1-BER-encoded PKCS#8 EncryptedPrivateKeyInfo structure.
+        ///   <para>
+        ///       -or-
+        ///   </para>
+        ///   The base-64 decoded contents of the PEM text from <paramref name="input" />
+        ///   indicate the key is for an algorithm other than the algorithm
+        ///   represented by this instance.
+        ///   <para>
+        ///       -or-
+        ///   </para>
+        ///   The base-64 decoded contents of the PEM text from <paramref name="input" />
+        ///   represent the key in a format that is not supported.
+        ///   <para>
+        ///       -or-
+        ///   </para>
+        ///   The algorithm-specific key import failed.
+        /// </exception>
+        /// <remarks>
+        ///   <para>
+        ///   The password bytes are passed directly into the Key Derivation Function (KDF)
+        ///   used by the algorithm indicated by <c>pbeParameters</c>. This enables compatibility
+        ///   with other systems which use a text encoding other than UTF-8 when processing
+        ///   passwords with PBKDF2 (Password-Based Key Derivation Function 2).
+        ///   </para>
+        ///   <para>
+        ///   The base-64 decoded contents of the PEM text only supports
+        ///   the binary (BER/CER/DER) encoding of EncryptedPrivateKeyInfo.
+        ///   </para>
+        ///   <para>
+        ///   Unsupported or malformed PEM-encoded objects will be ignored. If multiple supported PEM labels
+        ///   are found, an exception is raised to prevent importing a key when
+        ///   the key is ambiguous.
+        ///   </para>
+        ///   <para>This method supports the <c>ENCRYPTED PRIVATE KEY</c> PEM label.</para>
+        /// </remarks>
+        public override void ImportFromEncryptedPem(ReadOnlySpan<char> input, ReadOnlySpan<byte> passwordBytes)
+        {
+            PemKeyImportHelpers.ImportEncryptedPem<byte>(input, passwordBytes, ImportEncryptedPkcs8PrivateKey);
```

Comment:
Since ImportEncryptedPkcs8PrivateKey comes from AsymmetricAlgorithm, why do all of the key types need to implement this method?  Seems like it could just be the virtual implementation...

Reply:
@bartonjs that's a good point. [The API proposal][1] had all of them overriden. If we remove the `override`s from the RSA/DSA/ECDsa, do we need to go through another round of API Review?

[1]: https://github.com/dotnet/runtime/issues/31201#issue-558482299

Reply:
Nah, from the pure API perspective it was only the `virtual` that was interesting.  Removing overrides that don't yet actually exist is just an implementation detail.

Reply:
I'd probably just cut the sentence.  Assuming it's mirroring what I wrote on ImportPkcs8PrivateKey (assuming I wrote something like that), I was just saying "this doesn't understand PEM" :)